common/bin/gssetup.sh

```diff
@@ -56,77 +45,126 @@ usecase=""
 documentbase=""
 
 
+## ## ## ## ## ## ## ## ## ## ## ## ## ## ## ##
+# Verify readiness
+##
+
+# Verify current branch is not 'main'
+if [[ "${currentbranch}" == "main" ]]; then
+  echo
+  echo "  - - - - - - - - - - - - - - - - - - - - - - - - - - - -"
+  echo "  - Be sure to work in a branch other than 'main'."
+  echo "  -"
+  echo "  - Create a branch for your project and check it out,"
+  echo "  - then re-run this script."
+  echo "  - - - - - - - - - - - - - - - - - - - - - - - - - - - -"
+  echo
+  exit 1
+fi
+
+
+# Verify present working directory is correct for a getting started guide
+if [[ ! $PWD =~ kubernetes/start$ ]] && [[ ! $PWD =~ linux/start$ ]]; then
+  echo
+  echo "  - - - - - - - - - - - - - - - - - - - - - - - - - - - -"
+  echo "  - Your current working directory is:"
+  echo "  - '$PWD'"
+  echo "  -"
+  echo "  - Be sure you change to the 'kubernetes/start' or"
+  echo "  - 'linux/start' subdirectory, then execute this"
+  echo "  - script again."
+  echo "  - - - - - - - - - - - - - - - - - - - - - - - - - - - -"
+  echo
+  exit 2
+else
+  category=$(dirname $PWD)
```

Comment:
There is an internal Shell abbreviation which does the same, but without executing a command:

```suggestion
  category=${PWD%/*}
```

Reply:
Thank you.
I was torn on which approach to use.
The shell shortcut is clear enough to me, but I worried about others who might need to edit the script.
I think you've convinced me to switch this in the next revision.